puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/nightly_packaging.groovy

```diff
@@ -0,0 +1,42 @@
+def setup_nightly_build_environment(args) {
+    def commit_hash = ''
+    def project_name = args.project_name
+    def git_url = args.git_url
+    def git_ref = args.git_ref
+    def ruby_version = args.ruby_version ?: env.ruby_version
+
+    dir(project_name) {
+        // the full checkout() is necessary for checking out specific commits,
+        // as opposed to just branches with git()
+        checkout([
+            $class : 'GitSCM',
+            branches : [[name: git_ref]],
+            extensions: [[$class: 'CleanCheckout']],
+            userRemoteConfigs: [[url: git_url]]
+        ])
+    }
+    dir('foreman-packaging') {
```

Comment:
This and obal is only needed when building RPMs

Reply:
Sure, but currently all of our nightly packages build an RPM.

Reply:
We are currently only building RPMs with this pipeline.

Reply:
but you have `build_deb = true` for some jobs below, or am I missing something?

Reply:
True, but I'm always a fan of being explicit about what's there for which reason.

Reply:
> but you have `build_deb = true` for some jobs below, or am I missing something?

I probably could've been clearer. We are currently only building RPMs *within* the pipelines included in this PR. DEB builds are carried out in other jobs.

Reply:
gotcha